alignak/alignakobject.py

```diff
@@ -20,70 +20,161 @@
+    def __init__(self, params=None, parsing=True, debug=False):  # pylint: disable=W0613
+        """Initialize an Alignak base object
+
+        This method will create and set an object uuid if none exists. If no uuid still exists
+        then the object is considered as a fresh new object and all the properties are initialized
+        else the properties initialization step is skipped.
+
+        Properties initialization is done by creating a new attribute for each item defined in
+        the `properties`, `running_properties` and ` macros` dictionaries. The newly created
+        attribute is valued with the default value for the corresponding entry. If no default value
+        is defined then the attribute is set as None.
+
+        If some parameters are provided, the object attributes are updated with the corresponding
+        parameters value even if those attributes do not exist previously.
 
+        If debug is set, then this method will print information about the object properties
+
+        The parsing parameter indicates that the object is created from the configuration. As such,
+        if a property exists in the parameters the corresponding Property has its
+        `configuration_set` property set to True, else it remains False. This to allow knowing if
+        a property has been valued as default or from the configuration.
+
+        :param debug: print debug information about the object properties
+        :param params: parameters used to create the object
+        :param parsing: if True, initial creation, else, object unserialization
+        """
         if params is None:
-            return
-        all_props = {}
-        all_props.update(getattr(self, "properties", {}))
-        all_props.update(getattr(self, "running_properties", {}))
+            params = {}
+
+        if debug:
+            print('AlignakObject __init__: %s, %d params' %
+                  (self.__class__, len(params)))
+            print('AlignakObject __init__: %s, params list: %s' %
+                  (self.__class__, [key for key in params]))
+            print('AlignakObject __init__: %s, %d properties' %
+                  (self.__class__, len(self.properties)))
+            print('AlignakObject __init__: %s, properties list: %s' %
+                  (self.__class__, [key for key in self.properties]))
+            print('AlignakObject __init__: %d running properties' %
+                  len(getattr(self, "running_properties", {})))
+            print('AlignakObject __init__: %s, running properties list: %s' %
+                  (self.__class__, [key for key in self.running_properties]))
+
+        # Fresh new object or object update/unserialization?
+        if parsing:
+            self.uuid = uuid.uuid4().hex
+
+        # Create attributes for class properties, running properties and macros (to be confirmed?)
+        object_properties = {}
+        object_properties.update(getattr(self.__class__, "properties", {}))
```

Comment:
**Sure there is no problem**, because each class in the hierarchy updates its own properties from its parent class. As such, an Host has the properties of a SchedulingItem + Item + AlignakObject